NuRadioReco/detector/response.py

```diff
@@ -480,31 +480,35 @@ def get_time_delays(self):
         """ Get time delay from DB """
         return self.__time_delays
 
-    def _calculate_time_delay(self):
+    def calculate_time_delay(self, fmin=150*units.MHz, fmax=200*units.MHz):
         """
         Calculate time delay from phase of the stored complex response function.
         This is not the time delay which is stored in the DB and which is used in
         the `__init__()` to normalize the response function. Rather, its the remaining
         group delay.
 
-        The time delay is calculated as the mean between 195 and 205 MHz.
+        The time delay is calculated as the mean between fmin to fmax.
+
+        Parameters
+        ----------
+        fmin: float
+            lower frequency limit to use in the averaging of the group delay
+        fmax: float
+            upper frequency limit to use in the averaging of the group delay
 
         Returns
         -------
 
-        time_delay1 : float
-            The time delay at ~ 200 MHz
+        avg_delay : float
+            The avg group delay over the band fmin to fmax
         """
 
-        freqs = np.arange(50, 1200, 0.5) * units.MHz
-
-        response = self(freqs)
-        delta_freq = np.diff(freqs)
-        phase = np.angle(response)
-
-        time_delay = -np.diff(np.unwrap(phase)) / delta_freq / 2 / np.pi
-        mask = np.all([195 * units.MHz < freqs, freqs < 250 * units.MHz], axis=0)[:-1]
-        time_delay1 = np.mean(time_delay[mask])
+        freqs = np.linspace(fmin, fmax, 1000)
+        response = self(freqs) #make sure trigger = true does happen too
+        phase_angle = np.angle(response)
+        unwrapped = np.unwrap(phase_angle)
+        group_delays = -np.gradient(unwrapped) / (2 * np.pi * np.gradient(freqs))
+        avg_delay = np.mean( group_delays[np.logical_and(freqs>fmin, freqs<fmax)] )
```

Comment:
thinking about it - it is probably easier to create a frequency vector between `fmin` and `fmax` and not a larger one and than mask it.

Reply:
upos you are doing that already, hm, why the mask then?